Makefile

```diff
@@ -131,8 +131,10 @@ docker:
 	@git archive --format=tar.gz HEAD | docker build -t hypothesis/hypothesis:$(DOCKER_TAG) -
 
 .PHONY: docker-ws
-docker-ws:
+docker-ws: docker
+	docker tag hypothesis/hypothesis:dev hypothesis/hypothesis:latest 
```

Comment:
It feels a little wrong to me to apply `latest` tag to the `dev` image. For example could it somehow result in `dev` images accidentally getting published to Docker Hub as `latest`?

Could we instead make the `FROM` tag parameterisable in `streamer/Dockerfile`?

```Dockerfile
# h/streamer/Dockerfile
ARG TAG
FROM $TAG

...
```

Then in `Makefile`:

<pre><code># Makefile
.PHONY: docker-ws
docker-ws:
	docker build <b>--build-arg TAG=hypothesis/hypothesis:$(DOCKER_TAG)</b> -t hypothesis/hypothesis-ws:$(DOCKER_TAG) h/streamer </code></pre>

Based on this example: https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact

You can use envvars in `Dockerfile` too with `ENV`

Reply:
This is probably the crux of how to fix this